python_modules/dagster/dagster/components/core/component_tree.py

```diff
@@ -143,6 +143,58 @@ def get_direct_defs_dependents_of_component(
             ]
         )
 
+    def get_all_dependents_of_component(
+        self, defs_module_path: Path, component_path: ComponentPath
+    ) -> list[ComponentPath]:
+        """Returns a topologically sorted list of all components that depend on the given component,
+        starting from the given component and ending with root components which have no dependents.
+
+        For example, if component A depends on component B, and component B depends on component C,
+        then the load dependents of component C are [B, A].
+
+        Args:
+            defs_module_path: The defs module path used when dependencies were marked.
+            component_path: The component path to get dependents of.
+
+        Returns:
+            A topologically sorted list of all components that depend on the given component.
+        """
+
+        def _get_all_dependents_recursive(
+            canonical_path: tuple[str, Optional[Union[int, str]]],
+            visited: set[tuple[str, Optional[Union[int, str]]]],
+            result: list[ComponentPath],
+        ) -> None:
+            if canonical_path in visited:
+                return
+
+            visited.add(canonical_path)
+
+            direct_dependents = (
+                self._component_load_dependents_dict[canonical_path]
+                | self._component_defs_dependents_dict[canonical_path]
+            )
+
+            for dependent_file_path, dependent_instance_key in direct_dependents:
+                dependent_component_path = ComponentPath(
+                    file_path=Path(dependent_file_path).relative_to(defs_module_path),
+                    instance_key=dependent_instance_key,
+                )
+                if dependent_component_path not in result:
+                    result.append(dependent_component_path)
+
+            # Then recursively process each direct dependent (depth-first)
+            # This ensures topological ordering: direct dependents come before their dependents
+            for dependent_canonical_path in direct_dependents:
+                _get_all_dependents_recursive(dependent_canonical_path, visited, result)
+
+        canonical_target_path = _get_canonical_component_path(defs_module_path, component_path)
+
+        result = []
+        visited = set()
+        _get_all_dependents_recursive(canonical_target_path, visited, result)
+        return result
```

Comment:
when can you call this? any time or only after a full load? i think this touches on that idea from downstack about having a formal immutable finalized deps object that is set upon full load